src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -663,23 +748,29 @@ private void onSignal(
 
             switch ((int) signalId)
             {
-            case TOKEN_EXPIRED_SIGNAL:
-                onTokenExpiredSignal(signal);
+            case GRANT_VALIDATION_SIGNAL:
+                onGrantValidationSignal(signal);
                 break;
             default:
                 break;
             }
         }
 
-        private void onTokenExpiredSignal(
+        private void onGrantValidationSignal(
             SignalFW signal)
         {
-            final long delay = grant.expiresAt - System.currentTimeMillis();
+            final long now = System.currentTimeMillis();
+            long delay = grant.expiresAtMillis - now;
 
             if (delay > 0)
             {
-                this.expiryFuture = executor.schedule(delay, MILLISECONDS,
-                        targetRouteId, targetStreamId, TOKEN_EXPIRED_SIGNAL);
+                long nextSignalDelay = 0;
```

Comment:
This should be `delay`, not `0` right?
If so, then we can rename `delay` to `nextSignalDelay` and remove this line.